JSONObject.java

```diff
@@ -578,10 +581,10 @@ public double getDouble(String key) throws JSONException {
         Object object = this.get(key);
         try {
             return object instanceof Number ? ((Number) object).doubleValue()
-                    : Double.parseDouble((String) object);
+                    : new BigDecimal((String) object).doubleValue();
```

Comment:
Introducing a `BigDecimal` to parse a `double` will add too much overhead for no reason, if you try to represent a `double` from a `String` it will stay as is, if it was indeed a `double` then nothing will be gained or lost, you are only adding more GC overhead.

Same thing for your other `BigDecimal` introduction when working with `float` or `double`